### ISSUE TITLE
[Bug] Email not marked as read when inside thread

### ISSUE THREAD

**Issue by EvilVir** [bkaankose/Wino-Mail#242]
**Describe the bug**
If I have email's thread and select one of the messages it gets marked as read but it doesn't save and badge on the sidebar doesn't disappear. Next time I reopen the mailbox same email is again marked as unread.

**Expected behavior**
Emails should be properly marked as read.

**Screenshots**
<img width="527" alt="image" src="https://github.com/bkaankose/Wino-Mail/assets/2843963/636e354c-8b52-4d97-83df-f50e998e4638">

**Additional context**
Outlook.com mailbox, Wino 1.7.3 from Windows Store

**Comment by Tiktack** [comment]
Cannot re-produce, just tested 5 different threads. It works. I can admit there is some delay before bage updates, but it works.

**Comment by EvilVir** [comment]
I need to go out now but later I will record a video of this happening and send to you.

Have you checked exactly on Outlook.com mailbox? I feel like connector for Outlook is, in general, buggy compared to Gmail one.

**Comment by EvilVir** [comment]
Ok, here's short video:

https://www.icloud.com/iclouddrive/0afraqLaUwr5N1g357jOUyQTA#Recording_2024-06-20_132759

**Comment by bkaankose** [comment]
I agree that Outlook synchronizer is more buggy compared to Gmail one. Gmail has better API exposure, much easier to deal with to be honest...

Wino sends the request for marking the mail as read, applies a UI change (that's why you see it's marked as read in the mailing list), then triggers a quick synchronization to sync changes to acknowledge whether API responded to your 'mark as read' change. Badge is updated after this sync is done and your mail is marked as 'read' in the local database.

Probably something went wrong with the API call, and it's not reported back to Wino. It might be logged though. In the current version sharing logs is buggy, but it'll be fixed for the next release.

Could you manually share WinoDiagnostics.log file under this location:

C:\Users\YourUserName\AppData\Local\Packages\58272BurakKSE.WinoMailPreview_mhdqskaa8n2sj\LocalState

Don't forget to change the **YourUserName** part with your username. You can also delete this file (when the app is not running) and re-launch the app to have clear logs of this issue. I can investigate the issue based on this log.

**Comment by EvilVir** [comment]
[WinoDiagnostics.log](https://github.com/user-attachments/files/15920446/WinoDiagnostics.log)

Here you go :)

**Comment by bkaankose** [comment]
Looks like somehow marking that mail as read is missed during synchronization, but the folder's last sync identifier is updated to latest token. Unfortunately I couldn't detect anything from the logs.

I would try to replicate with different scenarios to see if I can reproduce it on my end.

**Comment by bkaankose** [comment]
New version v2.0.13.0 is released to Microsoft Store yesterday. Closing as fixed. If this is still reproducible, please reopen with steps.
